aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -213,7 +213,110 @@ The DI system is based on the DI system in ASP.NET Core. For more information, s
 
 ## Request a service in a component
 
-After services are added to the service collection, inject the services into the components using the [`@inject`](xref:mvc/views/razor#inject) Razor directive, which has two parameters:
+:::moniker range=">= aspnetcore-9.0"
+
+For injecting services into components, Blazor supports [constructor injection](#constructor-injection) and [property injection](#property-injection).
+
+### Constructor injection
+
+After services are added to the service collection, inject one or more services into components with constructor injection. The following example injects the `IDataAccess` service to obtain a list of actors for display by a component.
+
+`TheSunmakersCtorInjection.razor`:
+
+```razor
+@page "/the-sunmakers-ctor-injection"
+
+<PageTitle>The Sunmakers Ctor Injection</PageTitle>
+
+<h1>Doctor Who&reg;: The Sunmakers Actors (Villains)</h1>
+
+@if (actors != null)
+{
+    <ul>
+        @foreach (var actor in actors)
+        {
+            <li>@actor.FirstName @actor.LastName</li>
+        }
+    </ul>
+}
+
+<a href="https://www.doctorwho.tv">Doctor Who</a> is a
+    registered trademark of the <a href="https://www.bbc.com/">BBC</a>. 
+<a href="https://www.doctorwho.tv/stories/the-sunmakers">The Sunmakers</a>
+```
+
+`TheSunmakersCtorInjection.razor.cs`:
+
+```csharp
+using Microsoft.AspNetCore.Components;
+
+public partial class TheSunmakersCtorInjection(IDataAccess dataAccess)
```

Comment:
Instead of using a customer service, maybe just use a built-in one, like NavigationManager?

Reply:
Sure ... I was just mirroring the later example for property injection. Both of them can be simplified. I'll commit it first thing tomorrow morning if that's ok.